frontend/packages/kubevirt-plugin/src/k8s/wrapper/common/object-with-type-property-wrapper.ts

```diff
@@ -1,82 +1,86 @@
 import * as _ from 'lodash';
-import { ValueEnum } from '../../../constants';
+import { ObjectEnum } from '../../../constants';
+import { omitEmpty } from '../../../utils/basic';
 import { Wrapper } from './wrapper';
 
 export abstract class ObjectWithTypePropertyWrapper<
   RESOURCE,
-  TYPE extends ValueEnum<string>
-> extends Wrapper<RESOURCE> {
+  TYPE extends ObjectEnum<string>,
+  COMBINED_TYPE_DATA,
+  SELF extends ObjectWithTypePropertyWrapper<RESOURCE, TYPE, COMBINED_TYPE_DATA, SELF>
+> extends Wrapper<RESOURCE, SELF> {
   private readonly TypeClass: { getAll: () => TYPE[] | Readonly<TYPE[]> };
 
   private readonly typeDataPath: string[];
 
-  protected static defaultMergeWrappersWithType = <
-    A,
-    B extends ObjectWithTypePropertyWrapper<A, any>
-  >(
-    Clazz,
-    wrappers: B[],
-  ): B => {
-    const result = Wrapper.defaultMergeWrappers(Clazz, wrappers);
-    const lastWithType = _.last(wrappers.filter((wrapper) => wrapper && wrapper.getType()));
-
-    if (lastWithType) {
-      result.setType(lastWithType.getType(), result.getTypeData(lastWithType.getType()));
-    }
-    return result;
-  };
-
-  constructor(
-    data: RESOURCE,
-    opts: { initializeWithType?: TYPE; initializeWithTypeData?: any; copy?: boolean },
+  protected constructor(
+    data: RESOURCE | SELF,
+    copy = false,
     typeClass: { getAll: () => TYPE[] | Readonly<TYPE[]> },
     typeDataPath: string[] = [],
   ) {
-    super(data, opts);
+    super(data, copy);
     this.TypeClass = typeClass;
     this.typeDataPath = typeDataPath;
-
-    if (opts && opts.initializeWithType) {
-      const { initializeWithTypeData, initializeWithType, copy } = opts;
-
-      const resultTypeData = initializeWithTypeData
-        ? copy
-          ? _.cloneDeep(initializeWithTypeData)
-          : initializeWithTypeData
-        : this.getTypeData(initializeWithType);
-      this.setType(initializeWithType, resultTypeData);
-    }
   }
 
-  getType = (): TYPE =>
+  public getType = (): TYPE =>
```

Comment:
FYI, the default class member access modifier is `public`.

https://www.typescriptlang.org/docs/handbook/classes.html#public-by-default

Reply:
good to know, removed